proposals/public-suffix.md

```diff
@@ -0,0 +1,555 @@
+| net                | is a public suffix itself                        | Error                  |
+| foobar             | no matching suffix in PSL, assume 1-label suffix | Error                  |
+| net.foobar         | no matching suffix in PSL, assume 1-label suffix | net.foobar             |
+| 127.0.0.1          | IP address, IPv4                                 | Error                  |
+| [::1]              | IPv6 address                                     | Error                  |
+| EXAMPLE.NET        | uppercase                                        | example.net            |
+| .example.net       | dot in front                                     | example.net            |
+| example.net.       | dot in the end, this is an FQDN                  | example.net.           |
+| *.com              | contains invalid character `'*'`                 | Error                  |
+| github.io          | is a public suffix in the Private section        | Error                  |
+| github.io          | as above, with `excludePrivateSuffixes = true`   | github.io              |
+| foobar.github.io   | has a public suffix in the Private section       | foobar.github.io       |
+| foobar.github.io   | as above, with `excludePrivateSuffixes = true`   | github.io              |
+| مليسيا             | this is an IDN that is also a public suffix      | Error                  |
+| xn--mgbx4cd0ab     | as above, but Punycode                           | Error                  |
+| foo.مليسيا         | this is an IDN                                   | foo.مليسيا             |
+| foo.مليسيا         | as above, with `punycode = true`                 | foo.xn--mgbx4cd0ab     |
+| foo.xn--mgbx4cd0ab | this is an IDN, but Punycode                     | foo.مليسيا             |
+| foo.xn--mgbx4cd0ab | as above, with `punycode = true`                 | foo.xn--mgbx4cd0ab     |
+|                    | empty string                                     | Error                  |
+| .                  | no domain labels                                 | Error                  |
+| example..com       | contains an empty domain label                   | Error                  |
+
+#### 8. Batching
+
+For [Use Case #2](#2-group-domains-in-ui), function `getRegistrableDomains()` enables
+extensions to get multiple registrable domains with a single API call.
+
+The returned promises's resolved array should contain a `RegistrableDomains` object
+for each item in the input `domains` parameter, with the same ordering.
+
+An error during the lookup of any of the domain names does not cause the returned promise
+to be rejected. Instead, each such error is stored in the `error` key of the corresponding `RegistrableDomains` object for the domain name in question.
+
+### New Permissions
+
+| Permission Added | Suggested Warning |
+| ---------------- | ----------------- |
+| publicSuffix     | N/A               |
```

Comment:
This seems fine but it should explain why the browser is not more fingerprintable, for example, by arguing that the browser version is already available.